MdeModulePkg/Core/Dxe/Mem/Page.c

```diff
@@ -1706,14 +1720,84 @@ CoreInternalFreePages (
 
   IsGuarded = IsPageTypeToGuard (Entry->Type, AllocateAnyPages) &&
               IsMemoryGuarded (Memory);
+

```

Comment:
I recommend the logic added here be moved to an internal helper function to compute MemoryWrittable.

Reply:
Sure, I'll move that.

Reply:
Resolving as this is no longer relevant with the current code.